storage/integration_test/src/integration_test.cc

```diff
@@ -1038,6 +1039,103 @@ TEST_F(FirebaseStorageTest, TestDeleteWithMaxRetryDuration) {
 }
 #endif  // FIREBASE_PLATFORM_DESKTOP
 
+#if FIREBASE_PLATFORM_DESKTOP
+TEST_F(FirebaseStorageTest, TestPutFileWithMaxRetryDuration) {
+  // Enable retrying of future errors for testing. This will retry putfile when
+  // it is unable to read the local file.
+  bool old_value =
+      firebase::storage::internal::g_retry_failed_future_for_testing;
+  firebase::storage::internal::g_retry_failed_future_for_testing = true;
+
+  int shorter_duration_sec = 2;
+  int longer_duration_sec = 6;
+  SignIn();
+
+  std::string path = PathForResource() + kPutFileTestFile;
+  // Cloud Storage expects a URI, so add file:// in front of local
+  // paths.
+  std::string file_path = kFileUriScheme + path;
+
+  // Delete the local file first to guarantee it does not already exist.
+  std::remove(path.c_str());
```

Comment:
Make sure this works on Windows (it probably does, but just to be safe)

Reply:
std::remove should be supported on Windows. The only operating-system specific comment I found about the function is related to whether or not it will delete a file that is already opened on Windows
https://www.programiz.com/cpp-programming/library-function/cstdio/remove

If I do an automated integration test run will it run a windows version of the integration test? That seems like a reasonable way to be extra sure.

Reply:
Confirmed that this test worked on windows:
TestPutFileWithMaxRetryDuration passed in the following test run
https://github.com/firebase/firebase-cpp-sdk/runs/5419440986?check_suite_focus=true